src/tfr_control/src/arm_action_server.cpp

```diff
@@ -0,0 +1,103 @@
+/****************************************************************************************
+ * File:    arm_action_server.cpp
+ * Node:    arm_action_server
+ * 
+ * Purpose: This ActionServer is what handles the movement of the arm to given points.
+ *          It's tasked with interfacing with the move_group node to execute these
+ *          actions.
+ * 
+ *          This file includes <tfr_msgs/Arm.h>, which is one of seven headers
+ *          built by catkin from `tfr_msgs/action/Example.action`:
+ *              devel/include/tfr_msgs/ExampleAction.h
+ *              devel/include/tfr_msgs/ExampleActionFeedback.h
+ *              devel/include/tfr_msgs/ExampleActionGoal.h
+ *              devel/include/tfr_msgs/ExampleActionResult.h
+ *              devel/include/tfr_msgs/ExampleFeedback.h
+ *              devel/include/tfr_msgs/ExampleGoal.h
+ *              devel/include/tfr_msgs/ExampleResult.h
+ * 
+ ***************************************************************************************/
+
+#include <actionlib/server/simple_action_server.h>
+#include <tfr_msgs/ArmMoveAction.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+
+typedef actionlib::SimpleActionServer<tfr_msgs::ArmMoveAction> Server;
+
+moveit::planning_interface::MoveGroupInterface *move_group;
+const robot_state::JointModelGroup *joint_model_group;
+
+// This is the method that will be called when a client makes use
+// of this server. The provided goal is the "input"
+void execute(const tfr_msgs::ArmMoveGoalConstPtr& goal, Server* server)
+{
+    // Set up the joint space goal vector to travel to based on the input goal
+    // from the action server
+    std::vector<double> joint_group_positions(4);
+    joint_group_positions[0] = goal->pose[0];
+    joint_group_positions[1] = goal->pose[1];
+    joint_group_positions[2] = goal->pose[2];
+    joint_group_positions[3] = goal->pose[3];
+
+    // Set the current target
+    move_group->setJointValueTarget(joint_group_positions);
+    moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+
+    // Try to plan to the given target
+    bool success = (move_group->plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
```

Comment:
This is definitely optional, but when you have something as long as `moveit::planning_interface::MoveItErrorCode::SUCCESS` I think adding a `typedef` or `using` statement is perfectly reasonable.

Reply:
Thanks, that's a good idea: went ahead and implemented it in an upcoming commit.